diff --git a/plugins/context_reaction_block.inc b/plugins/context_reaction_block.inc
index abcafd8..29069d5 100644
--- a/plugins/context_reaction_block.inc
+++ b/plugins/context_reaction_block.inc
@@ -367,7 +367,8 @@ class context_reaction_block extends context_reaction {
     $context_blocks = &drupal_static('context_reaction_block_list');;
     $contexts = context_active_contexts();
     if (!isset($context_blocks)) {
-      $info = $this->get_blocks();
+      $info = $this->get_blocks(NULL, NULL, $this->rebuild_needed());
+      $this->rebuild_needed(FALSE);
       $context_blocks = array();
       foreach ($contexts as $context) {
         $options = $this->fetch_from_context($context);
